subsys/bluetooth/host/classic/a2dp.c

```diff
@@ -255,10 +255,12 @@ static int a2dp_set_config_ind(struct bt_avdtp *session, struct bt_avdtp_sep *se
 		}
 	}
 
-	ops = stream->ops;
-	if (*errcode == 0) {
-		if ((ops != NULL) && (stream->ops->configured != NULL)) {
-			stream->ops->configured(stream);
+	if (stream != NULL) {
```

Comment:
Isn't the core issue the fact that you're not returning failure from the function after line 247 the same way as you do in all earlier branches where `*errcode` gets set? Also, if you don't enter the branch on 235 then it seems like you should just directly set `*errcode = 0` and return success (`0`) from the function.

A separate note, which is not directly related to this PR or the issues it fixes: you should never have APIs which have `int` return type and return `-1` in case of failure. The reason is that the common convention is to use POSIX error numbers for the negative errors, and there's a risk that `-1` will be interpreted as a specific error rather than a generic error (e.g. in Zephyr `-1` is the same as `-EPERM`).

Reply:
I have noticed that. I will follow this issue.

> A separate note, which is not directly related to this PR or the issues it fixes: you should never have APIs which have `int` return type and return `-1` in case of failure. The reason is that the common convention is to use POSIX error numbers for the negative errors, and there's a risk that `-1` will be interpreted as a specific error rather than a generic error (e.g. in Zephyr `-1` is the same as `-EPERM`).

Reply:
Whatever the function returns from line 247, the stream pointer always needs to be checked. And `stream` pointer only valid in line 249~254. So, I set the `stream` pointer with initialization value `NULL`. And than check it before using it.

> Isn't the core issue the fact that you're not returning failure from the function after line 247 the same way as you do in all earlier branches where `*errcode` gets set? Also, if you don't enter the branch on 235 then it seems like you should just directly set `*errcode = 0` and return success (`0`) from the function.
>

Reply:
The stream pointer is only set inside the branch that starts at line 235, so it seems inefficient to try to do stuff with it outside of that branch. I.e. I'd just revert the branch condition and do an early return from the function and then you can have the setting of the pointer and using the pointer as part of the same branch.

Reply:
I move the code block into `if-else` when `stream` is not NULL.

Reply:
Thanks. Another issue, which seems equally valid even though Coverity didn't catch it: the function does `return (*errcode == 0) ? 0 : -1;` at the end, but it seems like you have branches (both with and without this PR) where `*errcode` might be uninitialized.

Reply:
`*errcode = 0` is at the head of the function.

Reply:
> `*errcode = 0` is at the head of the function.

Thanks. I missed it due to slightly unintuitive whitespace usage there (or rather lack thereof)